fog/view/server/tests/smoke_tests.rs

```diff
@@ -725,31 +558,38 @@ fn test_start_with_missing_range(logger: Logger) {
         expected_records.extend(records);
     }
 
-    let mut allowed_tries = 60usize;
-    loop {
-        let result = view_client.request(0, 0, Default::default()).unwrap();
-        if result.highest_processed_block_count == 15 && result.last_known_block_count == 15 {
-            break;
-        }
-
-        if allowed_tries == 0 {
-            panic!("Server did not catch up to database! highest_processed_block_count = {}, last_known_block_count = {}", result.highest_processed_block_count, result.last_known_block_count);
-        }
-        allowed_tries -= 1;
-        sleep(Duration::from_millis(1000));
-    }
-    assert_eq!(server.highest_processed_block_count(), 15);
+    mc_fog_view_server_test_utils::wait_for_highest_processed_and_last_known(view_client, 15, 15);
+
+    let highest_processed_block_count =
+        mc_fog_view_server_test_utils::get_highest_processed_block_count(store_servers);
+    assert_eq!(highest_processed_block_count, 15);
 
-    assert_e_tx_out_records_sanity(&mut view_client, &expected_records, &logger);
+    mc_fog_view_server_test_utils::assert_e_tx_out_records(view_client, &expected_records);
 }
 
 /// Test that view server behaves correctly when there is a missing range
 /// between two ingest invocations.
-#[test_with_logger]
-fn test_middle_missing_range_with_decommission(logger: Logger) {
+#[parameterized(
+one_store = { 1, 40 },
+multiple_stores = { 5, 8 },
+)]
+fn test_middle_missing_range_with_decommission() {
+    let (logger, _global_logger_guard) = create_app_logger(o!());
     let mut rng: StdRng = SeedableRng::from_seed([123u8; 32]);
-    let (db_context, server, mut view_client) = get_test_environment(512, logger.clone());
-    let db = db_context.get_db_instance();
+    const VIEW_OMAP_CAPACITY: u64 = 512;
+    const STORE_COUNT: usize = 5;
+    const BLOCKS_PER_STORE: u64 = 8;
+    let store_block_ranges =
+        mc_fog_view_server_test_utils::create_block_ranges(STORE_COUNT, BLOCKS_PER_STORE);
```

Comment:
:wrench: Similar to above not sure if a parameterized test was intended
```suggestion
fn test_middle_missing_range_with_decommission(store_count: usize, blocks_per_store: u64) {
    let (logger, _global_logger_guard) = create_app_logger(o!());
    let mut rng: StdRng = SeedableRng::from_seed([123u8; 32]);
    const VIEW_OMAP_CAPACITY: u64 = 512;
    let store_block_ranges =
        mc_fog_view_server_test_utils::create_block_ranges(store_count, blocks_per_store);
```

Reply:
Done.